### PR TITLE
Translate Notifications

### PR DESCRIPTION
**Pull request by kobergj**
Allows to translate notifications from `userlog` service. 

Based on https://github.com/owncloud/ocis/pull/5699

~DO NOT MERGE BEFORE BASE IS MERGED~

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Web-Tests-ocis-smoke-ocis-storage-2</strong>](https://drone.owncloud.com/owncloud/ocis/19996/63/1) failed. Further test are cancelled...

**Comment by kobergj** [comment]
@mmattel could you do your magic with the readme extension and the new envvar?

File: services/userlog/README.md [hunk 1]
```diff
@@ -31,3 +31,20 @@ The `userlog` service provides an API to retrieve configured events. For now, th
...
+
+## Translations
+
+The `userlog` service has embedded translations sourced via transifex to provide a basic set of translated languages. These embedded translation are available for all deployment scenarious. In addition, the service supports custom translations, though it is currently not possible to just add custom translations to embedded ones. If custom translation are configured, the embedded ones are not used. To configure custom translations, the `USERLOG_TRANSLATION_PATH` environment variable needs to point to a base folder that will further contain the translation files. This path must be available from all instances of the userlog service, a shared storage is recommended. Translation files must be of type [.po](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html#PO-Files) or [.mo](https://www.gnu.org/software/gettext/manual/html_node/Binaries.html). For each language, the filename needs to be `userlog.po` (or `userlog.mo`) and stored in a folder structure defining the language code. In general the path/name pattern for a translation file needs to be:
```
Comment (mmattel) [inline services/userlog/README.md:37]:
```suggestion
The `userlog` service has embedded translations sourced via transifex to provide a basic set of translated languages. These embedded translation are available for all deployment scenarios. In addition, the service supports custom translations, though it is currently not possible to just add custom translations to embedded ones. If custom translation are configured, the embedded ones are not used. To configure custom translations, the `USERLOG_TRANSLATION_PATH` environment variable needs to point to a base folder that will further contain the translation files. This path must be available from all instances of the userlog service, a shared storage is recommended. Translation files must be of type [.po](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html#PO-Files) or [.mo](https://www.gnu.org/software/gettext/manual/html_node/Binaries.html). For each language, the filename needs to be `userlog.po` (or `userlog.mo`) and stored in a folder structure defining the language code. In general the path/name pattern for a translation file needs to be:
```

File: services/userlog/README.md [hunk 2]
```diff
@@ -31,3 +31,20 @@ The `userlog` service provides an API to retrieve configured events. For now, th
...
+```
+
+The language-code pattern is composed as `language[_territory]` where  `language` is the base language and `_territory` is optional defining a country.
+
+As example, for the language `de_DE`, one needs to place the corresponding translation files to `{USERLOG_TRANSLATION_PATH}/de_DE/LC_MESSAGES/userlog.po`.
+
+### Translation Rules
+
+* If a requested language-code is not available, the service tries to fallback to the base language if available. As example, if `de_DE` is not available, the service tries to fall back to translations in `de` folder.
+* If the base language is also not available like when the language code is `de_DE` and neither `de_DE` nor the `de` folder is available, the service falls back to the systems default `en`, which is the source of the texts provided by the code.
```
Comment (mmattel) [inline services/userlog/README.md:50]:
```suggestion
*   If a requested language-code is not available, the service tries to fallback to the base language if available. As example, if `de_DE` is not available, the service tries to fall back to translations in `de` folder.
*   If the base language is also not available like when the language code is `de_DE` and neither `de_DE` nor the `de` folder is available, the service falls back to the systems default `en`, which is the source of the texts provided by the code.
```
make codacy happy

File: go.mod [hunk 3]
```diff
@@ -293,3 +294,5 @@ require (
 )
 
 replace github.com/cs3org/go-cs3apis => github.com/c0rby/go-cs3apis v0.0.0-20230110100311-5b424f1baa35
+
+replace github.com/leonelquinteros/gotext => github.com/kobergj/gotext v0.0.0-20230309141732-b909eb0b8956
```
Comment (micbar) [inline go.mod:298]: ok, did you open an upstream PR from that branch?

File: services/userlog/README.md [hunk 4]
```diff
@@ -31,3 +31,20 @@ The `userlog` service provides an API to retrieve configured events. For now, th
...
+
+## Translations
+
+The `userlog` service has embedded translations sourced via transifex to provide a basic set of translated languages. These embedded translation are available for all deployment scenarios. In addition, the service supports custom translations, though it is currently not possible to just add custom translations to embedded ones. If custom translation are configured, the embedded ones are not used. To configure custom translations, the `USERLOG_TRANSLATION_PATH` environment variable needs to point to a base folder that will further contain the translation files. This path must be available from all instances of the userlog service, a shared storage is recommended. Translation files must be of type [.po](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html#PO-Files) or [.mo](https://www.gnu.org/software/gettext/manual/html_node/Binaries.html). For each language, the filename needs to be `userlog.po` (or `userlog.mo`) and stored in a folder structure defining the language code. In general the path/name pattern for a translation file needs to be:
```
Comment (micbar) [inline services/userlog/README.md:37]:
```suggestion
The `userlog` service has embedded translations sourced via transifex to provide a basic set of translated languages. These embedded translations are available for all deployment scenarios. In addition, the service supports custom translations, though it is currently not possible to just add custom translations to embedded ones. If custom translations are configured, the embedded ones are not used. To configure custom translations, the `USERLOG_TRANSLATION_PATH` environment variable needs to point to a base folder that will further contain the translation files. This path must be available from all instances of the userlog service, a shared storage is recommended. Translation files must be of type [.po](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html#PO-Files) or [.mo](https://www.gnu.org/software/gettext/manual/html_node/Binaries.html). For each language, the filename needs to be `userlog.po` (or `userlog.mo`) and stored in a folder structure defining the language code. In general the path/name pattern for a translation file needs to be:
```

File: services/userlog/pkg/service/l10n/.tx/config [hunk 5]
```diff
@@ -0,0 +1,10 @@
+[main]
+host = https://www.transifex.com
+
+[o:owncloud-org:p:owncloud:r:ocis-userlog]
```
Comment (micbar) [inline services/userlog/pkg/service/l10n/.tx/config:4]: should be correct

File: services/userlog/pkg/service/l10n/locale/default.pot [hunk 6]
```diff
@@ -0,0 +1,82 @@
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
```
Comment (micbar) [inline services/userlog/pkg/service/l10n/locale/default.pot:6]: is `, fuzzy` a generated part?

**Review by micbar — COMMENTED** [review]
Templates and translation sync looks good so far.

@kobergj shouldn't we exclude the .pot file from committing? I thought the l10n-read job will create it

File: go.mod (hunk 3 above)
Comment (kobergj) [reply go.mod:298]: Not yet. Struggling with unit tests. Will come soon

**Review by micbar — APPROVED** [review]
LGTM, CI issues will be follow ups.

**Pull request merged**